src/validator.js

```diff
@@ -221,10 +221,10 @@ export default class Validator {
           return this.detectTypeFromLayout()
             .then((addonType) => {
               addonMetaData.type = addonType;
-              return Promise.resolve(addonMetaData);
+              return addonMetaData;
```

Comment:
I don't know if this clears it up: http://jsbin.com/delatenike/edit?html,js,console 

In a nutshell there's three things you can do in a `then`:

* return a promise. 
* return a value
* throw an error

This is a brilliant reference: http://pouchdb.com/2015/05/18/we-have-a-problem-with-promises.html - I've been re-reading it as I'd sniffed out the Promise-wrapping stuff but then realized that a lot of the times we use `Promise.resolve()` unnecessarily as returning a value is equivalent.

That said if you want a promise when there isn't one the `return Promise.resolve()` *is* the right thing to do. We do alot of that in the tests for example where we're stubbing promisey code.

I like the author's rule which is to always `return` or `throw` in a `then()` and everyhing will be fine.